lib/gollum-lib/file.rb

```diff
@@ -12,7 +12,9 @@ class << self
       #
       # query     - The String path to match.
       # entry     - The BlobEntry to check against.
-      def path_match(query, entry, global_match = false)
+      # global_match - If true, find a File matching path's filename, but not it's directory (so anywhere in the repo)
+      # sub_spaces   - GitHub compatibility: substitutes - for spaces when comparing filenames.
+      def path_match(query, entry, global_match = false, sub_spaces = false)
         query == ::File.join('/', entry.path)
```

Comment:
global_match and sub_spaces not implemented for Files, which I think is ok?

Reply:
Depends on github's behavior, and 4.x behavior I suppose?

Reply:
On 4.x both global_match and sub_spaces work for Files, too. But Github wikis at present doesn't allow uploading Files anyway -- you can only add pages. Would it be an acceptable divergence from 4.x not to implement compatibility for this?

Reply:
I'm now thinking we should also add compatibility for Files, but only if it's doable.

Reply:
Done in the refactor I've just pushed